src/partitioner/local/RDFParser.cpp

```diff
@@ -0,0 +1,422 @@
+ */
+
+void GetConfig::readConfigFile(string &configFile, int graphId)
+throw(std::runtime_error) {
+
+    struct stat fileStatus;
+    this->graphID = graphId;
+    ofstream file;
+
+    errno = 0;
+    if (stat(configFile.c_str(), &fileStatus) == -1) // ==0 ok; ==-1 error
+    {
+        if (errno == ENOENT)      // errno declared by include file errno.h
+            throw (std::runtime_error("Path file_name does not exist, or path is an empty string."));
+        else if (errno == ENOTDIR)
+            throw (std::runtime_error("A component of the path is not a directory."));
+        else if (errno == ELOOP)
+            throw (std::runtime_error("Too many symbolic links encountered while traversing the path."));
+        else if (errno == EACCES)
+            throw (std::runtime_error("Permission denied."));
+        else if (errno == ENAMETOOLONG)
+            throw (std::runtime_error("File can not be read\n"));
+    }
+
+    // Configure DOM parser.
+
+    m_ConfigFileParser->setValidationScheme(XercesDOMParser::Val_Never);
+    m_ConfigFileParser->setDoNamespaces(false);
+    m_ConfigFileParser->setDoSchema(false);
+    m_ConfigFileParser->setLoadExternalDTD(false);
+
+    try {
+
+
+        m_ConfigFileParser->parse(configFile.c_str());
+        DOMDocument *xmlDoc = m_ConfigFileParser->getDocument();
+
+
+        DOMElement *elementRoot = xmlDoc->getDocumentElement();
+        if (!elementRoot) throw (std::runtime_error("empty XML document"));
+
+
+        DOMNodeList *children = elementRoot->getChildNodes();
+        const XMLSize_t nodeCount = children->getLength();
+
+        for (XMLSize_t xx = 0; xx < nodeCount; ++xx) {
+            std::vector<long> authorsInArticle;
+            long articleID;
+            string web_address;
+            string journal_name;
+            string year;
+            string month;
+            string title;
+            string type;
+            string volume;
+
+
+            DOMNode *currentNode = children->item(xx);
+            if (currentNode->getNodeType() &&  // true is not NULL
+                currentNode->getNodeType() == DOMNode::ELEMENT_NODE) // is element
+            {
+
+                // Found node which is an Element. Re-cast node as element
+                DOMElement *currentElement
+                        = dynamic_cast< xercesc::DOMElement * >( currentNode );
+                if (XMLString::equals(currentElement->getTagName(), TAG_Article_Reference) ||
+                    XMLString::equals(currentElement->getTagName(), TAG_Book_Reference) ||
+                    XMLString::equals(currentElement->getTagName(), TAG_Thesis_Reference) ||
+                    XMLString::equals(currentElement->getTagName(), TAG_Book_Section_Reference) ||
+                    XMLString::equals(currentElement->getTagName(), TAG_Conference_Proceedings_Reference)) {
+
+
+                    if (XMLString::equals(currentElement->getTagName(), TAG_Article_Reference)) {
```

Comment:
Instead of these set of if statements, can't you have switch statement?

Reply:
Yes it's possible

Reply:
I tried to apply switch case statements here. But it cannot be done here.

Reply:
Ok